### PR TITLE
Allow different types of request bodies, fix #18

### PR DESCRIPTION
**Pull request by erikdubbelboer**

### CODE REVIEW

**Comment by erikdubbelboer** [comment]
Let me know what you think @cjoudrey

**Comment by cjoudrey** [comment]
I think you might be right here. Seems like 'form' was more of a convenience option, but that could probably be removed in favour of a more general option.

I don't mind :fire: 'form' entirely instead of deprecating it.

Can you add a test for 'body'? Also will body length header be automatically set? I forget.

File: gluahttp.go [hunk 1]
```diff
@@ -138,37 +138,40 @@ func (h *httpModule) doRequest(L *lua.LState, method string, url string, options
...
+			if _, ok := body.(lua.LString); ok {
+				req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+			}
+		}
 
+		switch reqBody := body.(type) {
 		case lua.LString:
-			body := reqForm.String()
+			body := reqBody.String()
 			req.ContentLength = int64(len(body))
```
Comment (erikdubbelboer) [inline gluahttp.go]: Length is still set.

Comment (cjoudrey) [inline gluahttp.go]: Oops! That'll teach me to review on a mobile device. :p

**Comment by erikdubbelboer** [comment]
I added tests and merged.

**Pull request merged**